core/src/mindustry/mod/ContentParser.java

```diff
@@ -327,6 +327,21 @@ public class ContentParser{
             readFields(consume, data);
             return consume;
         });
+        put(Team.class, (type, data) -> {
+            if(data.isString()){
+                Seq<Team> teams = new Seq<>(Team.baseTeams);
+                Team out = teams.find(t -> t.name.equals(data.asString()));
+                if(out == null) throw new IllegalArgumentException("Unknown team: " + data.asString());
+                return out;
+            }else if(data.isNumber()){
```

Comment:
Just get the team by ID, there are methods for that, this is extremely verbose and inefficient.